source/fundamentals/serialization/guid-serialization.txt

```diff
@@ -24,87 +24,104 @@ In this guide, you can learn how to serialize **globally unique identifiers**
+during encoding, different MongoDB drivers encoded GUIDs with different byte orders.
 
-In many MongoDB collections, all GUID fields use the same subtype of ``BsonBinaryData``.
-Some older collections, however, may contain some GUID fields that 
-use subtype 3 and others that use subtype 4. 
-To ensure that the driver serializes and deserializes all GUIDs correctly,  
-you should set the ``BsonDefaults.GuidRepresentationMode`` property to one of the 
-following ``GuidRepresentationMode`` values: 
+Use the following tabs to compare the ways in which different MongoDB language drivers
+encoded the preceding GUID to ``BsonBinaryData`` subtype 3:
 
-V2
-~~
+.. tabs::
 
-``GuidRepresentationMode.V2`` assumes that all GUIDs in a document use the same 
-``BsonBinaryData`` subtype. In this mode, GUID representation is 
-controlled by the reader or writer, not the serializer.
+   .. tab:: {+driver-short+}
+      :tabid: csharp
 
-``V2`` is the default ``GuidRepresentationMode``.
+      .. code-block:: csharp
 
-.. note::
-   
-   When version 3 of the {+driver-short+} is released, support for ``GuidRepresentationMode.V2``
-   will be removed from the driver and ``V3`` will become the default. 
+         33221100-5544-7766-8899-aabbccddeeff
 
-V3
-~~
+   .. tab:: PyMongo
+      :tabid: pymongo
 
-``GuidRepresentationMode.V3`` allows fields in the same document to use different 
-GUID formats. 
-In this mode, GUID representation is controlled at the property level by configuring the 
-serializer for each property.
+      .. code-block:: python
 
-To use ``GuidRepresentationMode.V3``, run the following line of code. You should run this 
-code during the bootstrapping phase of your application, before creating 
-a ``MongoClient`` object. 
+         00112233-4455-6677-8899-aabbccddeeff
 
-.. code-block:: csharp
+   .. tab:: Java Driver
+      :tabid: java
 
-   BsonDefaults.GuidRepresentationMode = GuidRepresentationMode.V3;
+      .. code-block:: java
 
-Running in ``V3`` mode changes the behavior of the driver in the following ways:
+         77665544-3322-1100-ffee-ddccbbaa9988
 
-- The ``BsonBinaryReader.ReadBinaryData()`` method ignores ``readerSettings.GuidRepresentation``
-- The ``BsonBinaryWriter.WriteBinaryData()`` method ignores ``writerSettings.GuidRepresentation``
-- The ``JsonReader.ReadBinaryData()`` method ignores ``readerSettings.GuidRepresentation``
-- ``JsonWriter`` ignores ``writerSettings.GuidRepresentation``
-- Calling the ``BsonBinaryData.ToGuid()`` method without the ``GuidRepresentation``
-  parameter works only on GUIDs of subtype 4.
+To standardize GUID byte order, we added ``BsonBinaryData`` subtype 4, which all
+MongoDB drivers encode in the same way. We recommend using 
+``BsonBinaryData`` subtype 4 for all new GUIDs.
 
-.. note::
+For a list of all ``BsonBinaryData`` subtypes, see the
+API documentation for the `BsonBinarySubType <{+new-api-root+}/MongoDB.Bson/MongoDB.Bson.BsonBinarySubType.html>`__
```

Comment:
Q: checking that this link will be added in v3

Reply:
yep, it will be broken until the 3.0 docs are published